python_modules/event-pipeline-demo/event_pipeline_demo/solids.py

```diff
@@ -0,0 +1,86 @@
+'''A fully fleshed out demo dagster repository with many configurable options.'''
+
+from dagster import check, InputDefinition, OutputDefinition, Result, SolidDefinition
+from dagster import DagsterUserCodeExecutionError
+
+
+from .configs import define_spark_config, parse_spark_config
+from .utils import run_spark_subprocess
+
+
+class SparkSolidError(DagsterUserCodeExecutionError):
+    pass
+
+
+def define_spark_transform_fn(context, inputs):
+    '''Define local Spark execution.
+
+    This function defines how we'll execute the Spark job in the local deployment configuration.
+    '''
+    # Extract parameters from config
+    (
+        main_class,
+        master_url,
+        deploy_mode,
+        application_jar,
+        spark_conf,
+        application_arguments,
+        spark_home,
+        spark_outputs,
+    ) = [
+        context.solid_config.get(k)
+        for k in (
+            'main_class',
+            'master_url',
+            'deploy_mode',
+            'application_jar',
+            'spark_conf',
+            'application_arguments',
+            'spark_home',
+            'spark_outputs',
+        )
+    ]
+
+    check.not_none_param(
+        spark_home,
+        'No spark home set. You must either provide spark_home or set $SPARK_HOME in your '
+        'environment',
+    )
+
+    deploy_mode = ['--deploy-mode', '{}'.format(deploy_mode)] if deploy_mode else []
+
+    spark_shell_cmd = (
+        ['{}/bin/spark-submit'.format(spark_home), '--class', main_class, '--master', master_url]
+        + deploy_mode
+        + parse_spark_config(spark_conf)
+        + [application_jar]
+        + ([application_arguments] if application_arguments else [])
+    )
+    system_context = context.get_system_context()
+    system_context.log.info("Running spark-submit: " + ' '.join(spark_shell_cmd))
+    retcode = run_spark_subprocess(spark_shell_cmd, system_context.log)
+
+    if retcode != 0:
+        raise SparkSolidError('Spark job failed')
+
+    for output_def in system_context.solid_def.output_defs:
+        yield Result(spark_outputs, output_def.name)
+
+
+def define_spark_solid(name, inputs, outputs, description=None):
```

Comment:
do we need a more specific name than `define_spark_solid` to distinguish this from a pyspark solid or a sparkr solid? very specifically this solid shells out to spark submit, which can be java or scala. (well any jvm language, but java or scala practically speaking)

Reply:
IMO a version of this will likely end up being the primary way that Spark jobs are invoked from dagster - Airflow calls this a `SparkSubmitOperator` https://github.com/apache/airflow/blob/master/airflow/contrib/operators/spark_submit_operator.py#L26, we could follow that naming?